pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/JsonExtractIndexTransformFunction.java

```diff
@@ -411,4 +407,18 @@ public String[][] transformToStringValuesMV(ValueBlock valueBlock) {
     }
     return _stringValuesMV;
   }
+
+  /**
+   * Lazily initialize _valueToMatchingDocsMap, so that map generation is skipped when filtering excludes all values
+   */
+  private Map<String, RoaringBitmap> getValueToMatchingDocsMap() {
+    if (_valueToMatchingDocsMap == null) {
```

Comment:
when this condition is true, does it mean the "filtering excludes all values"? Please clarify to match the method javadoc.

Reply:
It's a lazy initialization pattern, so the first time the method is called it will build the map. Subsequent calls will use the already built map. The pattern lets us avoid pre-building the map in transform function's `init` method since we might never need it, instead it is built when transforming the first `ValueBlock`